editor/components/post-taxonomies/index.js

```diff
@@ -18,24 +18,19 @@ import HierarchicalTermSelector from './hierarchical-term-selector';
 import FlatTermSelector from './flat-term-selector';
 import { getCurrentPostType } from '../../store/selectors';
 
-export function PostTaxonomies( { postType, taxonomies } ) {
+export function PostTaxonomies( { postType, taxonomies, taxonomyWrapper = identity } ) {
 	const availableTaxonomies = filter( taxonomies.data, ( taxonomy ) => includes( taxonomy.types, postType ) );
-
-	return (
-		<div>
-			{ availableTaxonomies.map( ( taxonomy ) => {
-				const TaxonomyComponent = taxonomy.hierarchical ? HierarchicalTermSelector : FlatTermSelector;
-				return (
-					<TaxonomyComponent
-						key={ taxonomy.slug }
-						label={ taxonomy.name }
-						restBase={ taxonomy.rest_base }
-						slug={ taxonomy.slug }
-					/>
-				);
-			} ) }
-		</div>
-	);
+	return availableTaxonomies.map( ( taxonomy ) => {
+		const TaxonomyComponent = taxonomy.hierarchical ? HierarchicalTermSelector : FlatTermSelector;
+		return taxonomyWrapper(
+			<TaxonomyComponent
+				key={ taxonomy.slug }
```

Comment:
I think there's some redundancy in `key` application. Technically this prop is not needed (nor used), *but* I do think it should be the responsibility of this component, not the callback for `taxonomyWrapper`, to apply the `key`. This could be done in one of two ways:

- Add a wrapper element to the return value of this `map` callback with the `key`
- Use `cloneElement` to assign `key` as a prop into the return value of `taxonomyWrapper`

Reply:
Yes, it makes sense to define the key here, I used a wrapper as it looks like the simplest approach.